.gitbook/5-oracles/1-operate-oracle-nodes/7-oracle-upgrade.md

```diff
@@ -0,0 +1,112 @@
+# Oracle Upgrade
+
+This document is about the whole process of oracle upgrade.
+
+## Oracle Upgrade Process
+
+All oracles registered in Panacea are forced to run oracle with the same unique ID stored in the oracle module.
+Since the new version (to be upgraded) of oracle will have a unique ID different from the existing one, the new unique ID should be registered in Panacea.
+This process is determined by on-chain governance, thus upgrade can be proposed by an `oracle-upgrade` proposal.
+
+### Submit Proposal for Oracle Upgrade
+
+Oracle upgrade can be proposed by submitting an `oracle-upgrade` proposal as shown below:
+
+```bash
+panacead tx gov submit-proposal oracle-upgrade \
+  --title "<proposal-title>" \
+  --description  "<proposal-description>" \
+  --upgrade-height <upgrade-target-height> \
+  --upgrade-unique-id "<oracle-unique-id>" \
+  --deposit "<deposit-in-umed>" \
+  --from "<proposer-key-name>" \
+  --chain-id "<chain-id>" \ 
+  --fees "1000000umed"
+```
+
+- title: The title of oracle upgrade proposal
+- description: The description of oracle upgrade proposal
+- upgrade-height: Target height to be upgraded
+- upgrade-unique-id: Unique ID of oracle to be upgraded
+- deposit: A deposit for proposal
+
+If the proposal passes, you can check the oracle upgrade information with the below CLI.
+
+```bash
+panacead q oracle oracle-upgrade-info
+```
+
+### Upgrade Oracle Node
+
+{% hint style="info" %}
+You can upgrade oracle any time after an `oracle-upgrade` proposal has passed (even before the upgrade target height is reached).
+However, since the new version of oracle can be active after the target height, the current version of oracle must be run before the target height.
+
+You do not have to stop the old version of oracle when upgrading to a new version of oracle.
```

Comment:
Does we need to guide change the `listen-addr`?

Reply:
I think we don't have to. If the same address is used, oracle daemon returns error and exit.
And I expect that they shut down the old version of oracle and start the new one after upgrade.